lib/bluetooth_low_energy/aws_ble_gatt.c

```diff
@@ -740,302 +555,159 @@ static void vIndicationSentCb( uint16_t usConnId,
-
-                    if( ( *ppxService )->pxCharacteristics != NULL )
-                    {
-                        memset( ( *ppxService )->pxCharacteristics, 0, sizeof( BLECharacteristic_t ) * xNbCharacteristic );
-
-                        for( xCharId = 0; xCharId < xNbCharacteristic; xCharId++ )
-                        {
-                            if( xNumDescrsPerChar[ xCharId ] > 0 )
-                            {
-                                ( *ppxService )->pxCharacteristics[ xCharId ].pxDescriptors = pvPortMalloc( sizeof( BLECharacteristicDescr_t * ) * xNumDescrsPerChar[ xCharId ] );
-
-                                if( ( *ppxService )->pxCharacteristics[ xCharId ].pxDescriptors == NULL )
-                                {
-                                    xStatus = eBTStatusNoMem;
-                                    break;
-                                }
-                                else
-                                {
-                                    memset( ( *ppxService )->pxCharacteristics[ xCharId ].pxDescriptors, 0, sizeof( BLECharacteristicDescr_t * ) * xNumDescrsPerChar[ xCharId ] );
-                                }
-                            }
-                            else
-                            {
-                                ( *ppxService )->pxCharacteristics[ xCharId ].pxDescriptors = NULL;
-                            }
-                        }
-                    }
-                    else
-                    {
-                        xStatus = eBTStatusNoMem;
-                    }
-                }
-
-                if( ( xNbDescriptors > 0 ) && ( xStatus == eBTStatusSuccess ) )
-                {
-                    ( *ppxService )->pxDescriptors = pvPortMalloc( sizeof( BLECharacteristicDescr_t ) * xNbDescriptors );
-
-                    if( ( *ppxService )->pxDescriptors != NULL )
-                    {
-                        memset( ( *ppxService )->pxDescriptors, 0, sizeof( BLECharacteristicDescr_t ) * xNbDescriptors );
-                    }
-                    else
-                    {
-                        xStatus = eBTStatusNoMem;
-                    }
-                }
-
-                if( ( xNbIncludedServices > 0 ) && ( xStatus == eBTStatusSuccess ) )
-                {
-                    ( *ppxService )->pxIncludedServices = pvPortMalloc( sizeof( BLEService_t * ) * xNbIncludedServices );
-
-                    if( ( *ppxService )->pxIncludedServices != NULL )
-                    {
-                        memset( ( *ppxService )->pxIncludedServices, 0, sizeof( BLEService_t * ) * xNbIncludedServices );
-                    }
-                    else
-                    {
-                        xStatus = eBTStatusNoMem;
-                    }
-                }
-            }
-        }
+    if( pxNewElem != NULL )
```

Comment:
Null check not required

Reply:
Thank you!

Reply:
You actually need it because of Malloc" pxNewElem = pvPortMalloc( sizeof( BLEServiceListElement_t ) );
But there was a bug where memset was done before the check